### PR TITLE
Fix history simple gd

### PR DESCRIPTION
**Pull request by mstechly**

### CODE REVIEW

File: src/python/zquantum/optimizers/simple_gradient_descent.py [hunk 1]
```diff
@@ -55,6 +55,8 @@ def _minimize(
         for _ in range(self.number_of_iterations):
             gradients = cost_function.gradient(current_parameters)
             current_parameters = current_parameters - (self.learning_rate * gradients)
+            if keep_history == True:
```
Comment (brian-dellabetta) [inline src/python/zquantum/optimizers/simple_gradient_descent.py:58]: the CI pipeline doesn't like this line, i think it just needs to be `if keep_history:`

Comment (brian-dellabetta) [reply src/python/zquantum/optimizers/simple_gradient_descent.py:58]:
```suggestion
            if keep_history:
```

**Review by brian-dellabetta — APPROVED** [review]

**Comment by mstechly** [comment]
@brian-dellabetta I've updated tests and docs, please let me know if it looks ok!

File: src/python/zquantum/optimizers/simple_gradient_descent.py [hunk 2]
```diff
@@ -42,11 +42,20 @@ def _minimize(
         Finds the parameters which minimize given cost function, by trying all
         the parameters from the provided list of points.
 
+        Note:
+            This optimizer does not require evaluation of the cost function,
+            but relies only on gradient evaluation. This means, that if we want to
+            keep track of values of the cost functions for each iteration, we
+            need to perform extra evaluations. Therefore using `keep_history=True`
+            will increase add extra evaluations that are not necessary for
```
Comment (dexter2206) [inline src/python/zquantum/optimizers/simple_gradient_descent.py:50]: I'm not sure if this sentence is correct. Shouldn't it be "Therefore using `keep_history=True` will ~increase~ add extra evaluations that are not necessary for the optimization process ~itslelf~ itself?

**Review by dexter2206 — CHANGES_REQUESTED** [review]
From the code perspective, it looks ok, but I recommend some minor adjustments to the docstring.

File: src/python/zquantum/optimizers/simple_gradient_descent.py (hunk 2 above)
Comment (mstechly) [reply src/python/zquantum/optimizers/simple_gradient_descent.py:50]: Will agree with the first point, but I really liked the spelling of "itslelf" :P

Comment (mstechly) [reply src/python/zquantum/optimizers/simple_gradient_descent.py]: Fixed in https://github.com/zapatacomputing/z-quantum-optimizers/pull/79/commits/a88fde62b7cbbf261056b185e6375f1c9b372543

**Review by dexter2206 — APPROVED** [review]

**Pull request merged**